contracts/dtoken/src/common.rs

```diff
@@ -0,0 +1,35 @@
+use near_sdk::Balance;
+use crate::*;
+
+#[near_bindgen]
+impl Contract {
+
+    pub fn get_exchange_rate(&self, underlying_balance: Balance) -> Balance {
+        let mut exchange_rate: u128;
```

Comment:
@pahomas precondition approach is more Rust way and make function easier to read (as for me)
so this could could be something like this

```
if self.token.total_supply > 0 {
  return self.initial_exchange_rate;
}

return (underlying_balance + self.total_borrows - self.total_supplies) / self.token.total_supply;
```